frontend/components/UpgradeURD.tsx

```diff
@@ -0,0 +1,319 @@
+import React, { useContext, useEffect, useState } from 'react';
+import {
+  Box,
+  Button,
+  Flex,
+  Step,
+  StepDescription,
+  StepIndicator,
+  StepNumber,
+  Stepper,
+  StepSeparator,
+  StepStatus,
+  StepTitle,
+  Text,
+  useColorModeValue,
+  useSteps,
+  useToast,
+  VStack,
+} from '@chakra-ui/react';
+import { WalletContext } from '@/components/wallet/WalletContext';
+import { ethers } from 'ethers';
+import LSP1GraveForwarderAbi from '@/abis/LSP1GraveForwarder.json';
+import { LSP1GraveForwarder } from '@/contracts';
+import { formatAddress } from '@/utils/tokenUtils';
+import { FaCheckCircle } from 'react-icons/fa';
+import {
+  resetLSPDelegates,
+  setForwarderAsLSPDelegate,
+  updateBECPermissions,
+} from '@/utils/urdUtils';
+
+const initialLeavingSteps = [
+  {
+    title: 'Give your 🆙 necessary permissions',
+    instructions: 'Give permissions to your Browser Extension Controller.',
+    instructions2: { text: 'This can also be done manually.', address: null },
+    completeText: { text: 'PERMISSION SET', address: null },
+    complete: false,
+  },
+  {
+    title: 'Reset previous LSP 7 and LSP8 delegate permissions',
+    completeText: { text: '', address: null },
+    complete: false,
+  },
+  {
+    title: 'Migrate your GRAVE to the new forwarder',
+    completeText: { text: '', address: null },
+    complete: false,
+  },
+  {
+    title: 'Direct all 🆙 spam to the GRAVE',
+    completeText: { text: 'SPAM IS DEAD', address: null },
+    complete: false,
+  },
+];
+
+export const UpgradeURD = ({
+  oldForwarderAddress,
+}: {
+  oldForwarderAddress: string;
+}) => {
+  const walletContext = useContext(WalletContext);
+  const bgColor = useColorModeValue('light.green.brand', 'dark.purple.200');
+  const provider = new ethers.providers.Web3Provider(window.lukso);
+  const signer = provider.getSigner();
+  const { account, networkConfig, setURDLsp7, setURDLsp8, mainUPController } =
+    walletContext;
+
+  const [isSubmitting, setIsSubmitting] = useState<boolean>(false);
+  const [leaveSteps, setLeaveSteps] = React.useState([...initialLeavingSteps]);
+  const [leavingStep, setLeavingStep] = useState<number>(-1);
+
+  const { activeStep: activeLeavingStep, setActiveStep: setActiveLeavingStep } =
+    useSteps({
+      index: -1,
+      count: leaveSteps.length,
+    });
+
+  const toast = useToast();
+
+  const migrateGraveToNewForwarder = async () => {
+    const oldForwarder = new ethers.Contract(
+      oldForwarderAddress,
+      LSP1GraveForwarderAbi.abi,
```

Comment:
if it is an old forwarder, the abi will not be the same

Reply:
We can store multiple ABIs once we make a change to getGrave